app/app.go

```diff
@@ -970,6 +970,7 @@ func (app *App) ProcessProposalHandler(ctx sdk.Context, req *abci.RequestProcess
 	} else if !bytes.Equal(app.optimisticProcessingInfo.Hash, req.Hash) {
 		app.optimisticProcessingInfo.Aborted = true
 	}
+	metrics.SetOptimisticProcessingCounter(!app.optimisticProcessingInfo.Aborted)
```

Comment:
why here? instead of in FinalizeBlocker

Reply:
Also shouldn't we increment the counter each time optimistic processing is elligible? Why setting a counter ?

Reply:
looks like the helper DOES increment a counter under the hood, maybe better to rename the helper to something like `IncrementOptimisticProcessingCounter`. Metric location also depends on whether we want to track the number of times optimistic proicessing is attempted vs succeeded. Since in finalize blocker we branch based on that, so if we want to track each metric separately then finalizeblocker is probably a better place to track

Reply:
Yeah, I think tracking optimistic processing success has more values, having both is even better